dynamometer-workload/src/main/java/com/linkedin/dynamometer/workloadgenerator/audit/AuditLogDirectParser.java

```diff
@@ -63,6 +64,7 @@ public AuditReplayCommand parse(Text inputLine, Function<Long, Long> relativeToA
     String auditMessageSanitized = m.group(2).replace("(options=", "(options:");
     Map<String, String> parameterMap = AUDIT_SPLITTER.split(auditMessageSanitized);
     return new AuditReplayCommand(relativeToAbsolute.apply(relativeTimestamp),
+        SPACE_SPLITTER.split(parameterMap.get("ugi")).iterator().next(),
```

Comment:
more of a question, does this imply there can be multiple values for ugi, and here we only take the first? I wonder why there can be multiple of them, and why taking just first is enough?

Reply:
Sort of. Here are a few example UGI strings:
```
ugi=user (auth:TOKEN) via user/host@REALM (auth:TOKEN)
ugi=user (auth:KERBEROS)
ugi=user@REALM (auth:TOKEN)
ugi=user/host@REALM (auth:TOKEN)
```
Taking the first is enough because the second is used if the UGI is proxied; the first user is the effective user which is all we care about.

However this made me realize that this current logic is not sufficient to cover the last two cases. Maybe it would be best to do a regex match for the first character string up until whitespace, `/`, or `@`. I will update accordingly.